controllers/storageclassrequest/storageclassrequest_controller_test.go

```diff
@@ -32,26 +33,74 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	pgAutoscaleMode = "pg_autoscale_mode"
+	pgNum           = "pg_num"
+	pgpNum          = "pgp_num"
+	namespaceName   = "test-ns"
+	deviceClass     = "ssd"
+	kind            = "StorageProfile"
+)
+
 var fakeStorageProfile = &v1.StorageProfile{
+	TypeMeta: metav1.TypeMeta{Kind: kind},
 	ObjectMeta: metav1.ObjectMeta{
 		Name:      "medium",
-		Namespace: "test-ns",
+		Namespace: namespaceName,
+	},
+	Spec: v1.StorageProfileSpec{
+		DeviceClass: deviceClass,
+	},
+}
+
+var validStorageProfile = &v1.StorageProfile{
+	TypeMeta: metav1.TypeMeta{Kind: kind},
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "valid",
+		Namespace: namespaceName,
 	},
 	Spec: v1.StorageProfileSpec{
-		DeviceClass: "ssd",
+		DeviceClass: deviceClass,
+		BlockPoolConfiguration: v1.BlockPoolConfigurationSpec{
+			Parameters: map[string]string{
+				pgAutoscaleMode: "on",
+				pgNum:           "128",
+				pgpNum:          "128",
+			},
+		},
 	},
+	Status: v1.StorageProfileStatus{Phase: ""},
+}
+
+var rejectedStorageProfile = &v1.StorageProfile{
```

Comment:
rejectedStorageProfile -> invalidStorageProfile sounds good WDYT?

Reply:
I wanted to emphasize the fact that invalid SPs are defined as ones that have their phase set to `Rejected` and are in essence rejected.

Reply:
Resolving as per @rchikatw's manual confirmation